File: applicatie/src/main/java/com/vdzon/administratie/gebruiker/GebruikerResource.java
package com.vdzon.administratie.gebruiker;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;


public class GebruikerResource {

    @Inject
    public GebruikerResource(GebruikerService gebruikerService) {
        Spark.post("/rest/gebruiker/", gebruikerService::postGebruiker, JsonUtil.json());
        Spark.put("/rest/gebruiker/", gebruikerService::postGebruiker, JsonUtil.json());
        Spark.delete("/rest/gebruiker/:uuid", gebruikerService::removeGebruiker, JsonUtil.json());
        Spark.post("/rest/gebruiker/updatePassword/:uuid/:newPassword", gebruikerService::updatePassword, JsonUtil.json());
    }


}

File: rest/impl/src/main/java/com/vdzon/administratie/rest/gebruiker/GebruikerResource.kt
package com.vdzon.administratie.rest.gebruiker

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject


class GebruikerResource
@Inject
constructor(gebruikerService: GebruikerService) {
    init {
        Spark.post("/rest/gebruiker/", Route { req, res -> gebruikerService.postGebruiker(req, res) }, JsonUtil.json())
        Spark.put("/rest/gebruiker/", Route { req, res -> gebruikerService.postGebruiker(req, res) }, JsonUtil.json())
        Spark.delete("/rest/gebruiker/:uuid", Route { req, res -> gebruikerService.removeGebruiker(req, res) }, JsonUtil.json())
    }


}
